### ISSUE TITLE
Add support for not contains json matcher

### ISSUE THREAD

**Issue by Crow-EH** [ekino/veggies#42]
Hi everyone.

I created this issue instead of a PR to talk about it with you first. I'm still 100% motivated to code it myself though.

Here's the issue: Today we can't make an **explicit**, **concise** and **human readable** test that a JSON field doesn't contain a string.

How about adding a way to negate matchers or add new negated matchers `not contain` / `not contains` ? (or even `!contains`)

# Example / Use case
Let's say I want to test this object : 
```
{
  a: string
  b: string
}
```

Additionally `a` and `b` :
- have very, very **long values** (e.g. URLs with many query params, tokens, etc.)
- might **evolve many times** in the project lifecycle (e.g. by adding params regularly)

=> **I don't want to use the `equals` matcher** because I'll have to update every tests every time, plus my tests would be over verbose.

Now let's pretend that I have to develop a new feature where:
- I need to add a new `foo` query param only for `a`
- But absolutely not for `b` (i.e. they have a similar method of generation and errors could be made)

I'd want to test that :
- `a` contains `foo`
- `b` does not contain `foo`

And here's what it would look like today :
```
And json response should match
    | field | matcher  | value         |
    | a     | contains | foo           |
    | b     | match    | ^((?!foo).)*$ |
```

The only way to do the second test is to match the value against `^((?!foo).)*$` with `match`, which is not really human readable and more error prone than a simple `not contains`.

The negated contains would look like this:
```
And json response should match
    | field | matcher      | value |
    | a     | contains     | foo   |
    | b     | not contains | foo   |
```

WDYT @plouc @leguellec ?

# TL;DR
Need to test if a value is not contained by another one, but regexps are not human friendly :

![tenor](https://user-images.githubusercontent.com/3366767/59781371-f3e63200-92bb-11e9-9619-54ae2244a2f6.gif)

**Comment by Crow-EH** [comment]
Here's a PR.

Not sure if I did everything right, so don't hesitate to roast me. 😄

Thank you !

**Comment by plouc** [comment]
Maybe we could use `does not` or `doesn't` instead as gherkin is supposed to be human readable, this way we'll have something like `props | doesn't contain | something`.

Also I think we could support this for all matchers, what do you think?

**Comment by Crow-EH** [comment]
Good point!
I'll rename it and extend to every matchers.

**Comment by plouc** [comment]
perfect, thanks!

**Comment by Crow-EH** [comment]
Done !

I added support for every matchers negation with: `!`, `not`, `does not`, `doesn't`, `is not`, `isn't`.
